File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.java
class Solution {
    public int numberOfArrays(int[] diff, int lower, int upper) {
        long pre = 0,           // running sum of differences
                minPre = 0,     // min prefix sum seen
                maxPre = 0;     // max prefix sum seen

        for (int d : diff) {
            pre += d;                   // extend prefix by current difference
            minPre = Math.min(minPre, pre);
            maxPre = Math.max(maxPre, pre);
        }

        long low = lower - minPre;
        long high = upper - maxPre;

        // number of integer values in [low, high], or 0 if empty
        return (int) Math.max(0, high - low + 1);
    }
}

File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.py
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        #InBuilt PreSum function with first value 0 
        pre = list(accumulate(differences, initial=0)) 

        return max(0, (upper - lower) - (max(pre) - min(pre)) + 1)

        